### ISSUE TITLE
Distance metric does not match the paper

### ISSUE THREAD

**Issue by YoucanBaby** [fudoki-hku/FUDOKI#7]
Thanks for the great work.

I have a small question about the code.

The distance metric in code:

    def metric(self, z):
        z_flattened = z.view(-1, z.shape[-1])
        z = F.normalize(z, p=2, dim=-1)
        z_flattened = F.normalize(z_flattened, p=2, dim=-1)
        embedding = F.normalize(self.embedding.weight, p=2, dim=-1)
        d = (torch.sum(z_flattened ** 2, dim=1, keepdim=True) + \
            torch.sum(embedding**2, dim=1) - 2 * \
            torch.einsum('bd,dn->bn', z_flattened, torch.einsum('n d -> d n', embedding))) ** 2
        return d

[https://github.com/fudoki-hku/FUDOKI/blob/8bb08e70a3ad6984b6968ba6fb594c29a7241a68/flow_matching/path/mixture.py#L146](url)

It seems the fourth power of the Euclidean distance. 

But the paper report that the metric is Euclidean distance. 

Why implement the code like this?

**Comment by haotiansun14** [comment]
My educated guess is that the authors of FUDOKI imported the metrics used for image generation in https://arxiv.org/pdf/2412.03487 (check App. E4, eq 108 with a = 0.9,c = 3, and lp = 4). But as this setting was originally proposed for images, could the authors please provide any insights into this choice? Thank you very much!

**Comment by jinjinw** [comment]
My apologies for the late reply.

Thank you for pointing this out. You're correct in your observation. This implementation is inherited from https://arxiv.org/pdf/2412.03487, which uses lp=4 for image generation. We found that this setting worked well in practice and did not explore other ablation choices for the metric calculation. It would be interesting to explore other metric calculation methods in future research.
